Base/QTApp/qSlicerApplicationHelper.txx

```diff
@@ -48,6 +49,58 @@
 namespace
 {
 
+/// Event filter that allows to move a widget by clicking anywhere on it.
+/// It also removes the WindowStaysOnTopHint so that the window is not forced to
+/// appear above all other applications.
+class DraggableWidgetEventFilter : public QObject
+{
+public:
+  void setWidget(QWidget* w)
+    {
+    this->Widget = w;
+    }
+protected:
+  bool eventFilter(QObject* obj, QEvent* event) override
+    {
+    if (event->type() == QEvent::MouseButtonPress && this->Widget)
+      {
+      QMouseEvent* mouseEvent = static_cast<QMouseEvent*>(event);
+      this->PressPosition = mouseEvent->pos();
+      this->Dragging = true;
+      if (this->DisableTopMost)
+        {
+        this->Widget->setWindowFlags(this->Widget->windowFlags() & ~Qt::WindowStaysOnTopHint);
```

Comment:
To ensure the "widget" is in the front, what about setting `Qt::WindowActive`[^1] in additional of removing `Qt::WindowStaysOnTopHint`

Alternatively, calling `raise`[^2] may be more suitable.

[^1]: https://doc.qt.io/qt-5/qt.html#WindowState-enum
[^2]: https://doc.qt.io/qt-5/qwidget.html#raise

Reply:
The topmost attribute is removed when the user clicks on the widget, so at that point the widget is already at the top. After that we don't want to force the widget to the top. So, I'm not sure why would we need to raise or activate.

Reply:
This was instead of the call to `show()` below which seems to bring back the widget to the top

Reply:
To allow click-and-dragging to work, we need to keep the window displayed at the top. The flag is changed so that when the user stopped moving the window and goes to another application the window will not stay on top anymore. The user click-and-dragging the window is a hint that the window bothers the user.

Reply:
Agreed, unsetting `WindowStaysOnTopHint` is key.

To clarify, I was initially thinking that instead of calling `show()`, setting the window state to active was more appropriate. A quick experiment demonstrated this was not the case.